src/test/storage/chunk_encoder_test.cpp

```diff
@@ -0,0 +1,119 @@
+#include <cstdint>
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "base_test.hpp"
+#include "gtest/gtest.h"
+
+#include "all_type_variant.hpp"
+#include "storage/base_value_column.hpp"
+#include "storage/chunk.hpp"
+#include "storage/chunk_encoder.hpp"
+#include "storage/encoded_columns/base_encoded_column.hpp"
+#include "storage/encoded_columns/encoded_columns.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+class ChunkEncoderTest : public BaseTest {
```

Comment:
Will "recompression" be a follow-up PR? I was expecting to see it here as the input column type should be relatively irrelevant due to the iterables, right?

Reply:
Yes that’s true. I’ve become a bit cautious about create too large pull requests and because I don’t need at the moment. I left it out.